ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java

```diff
@@ -172,32 +168,26 @@ public static BeaconStateWithCache get_initial_beacon_state(
     }
     List<List<Integer>> shuffling =
         get_shuffling(seed, state.getValidator_registry(), shuffling_epoch);
-    UnsignedLong offset = slot.mod(UnsignedLong.valueOf(SLOTS_PER_EPOCH));
-    UnsignedLong committees_per_slot =
-        committees_per_epoch.dividedBy(UnsignedLong.valueOf(SLOTS_PER_EPOCH));
+    long offset = slot % SLOTS_PER_EPOCH;
+    long committees_per_slot = committees_per_epoch / SLOTS_PER_EPOCH;
     // TODO: Double check the order of operations here. The spec may be ambiguous.
-    UnsignedLong slot_start_shard =
-        shuffling_start_shard
-            .plus(committees_per_slot)
-            .times(offset)
-            .mod(UnsignedLong.valueOf(Constants.SHARD_COUNT));
+    long slot_start_shard =
+        shuffling_start_shard + committees_per_slot * offset % Constants.SHARD_COUNT;
+
     ArrayList<CrosslinkCommittee> crosslink_committees_at_slot = new ArrayList<>();
-    for (long i = 0; i < committees_per_slot.longValue(); i++) {
+    for (long i = 0; i < committees_per_slot; i++) {
       CrosslinkCommittee committee =
           new CrosslinkCommittee(
-              committees_per_slot
-                  .times(offset)
-                  .plus(UnsignedLong.valueOf(i))
-                  .mod(UnsignedLong.valueOf(Constants.SHARD_COUNT)),
-              shuffling.get(toIntExact(slot_start_shard.longValue() + i) % Constants.SHARD_COUNT));
+              committees_per_slot * offset + i % Constants.SHARD_COUNT,
```

Comment:
I don't believe this should be modded by SHARD_COUNT, according to the spec. OOO looks good here though.